cpp/ycm/versioning.h

```diff
@@ -15,8 +15,17 @@
 // You should have received a copy of the GNU General Public License
 // along with YouCompleteMe.  If not, see <http://www.gnu.org/licenses/>.
 
+#ifndef VERSIONING_H_EEJUU0AH
+#define VERSIONING_H_EEJUU0AH
+
+#if !defined( YCMD_CORE_VERSION )
+  #define YCMD_CORE_VERSION 0
```

Comment:
Add a comment above this stating that the true value is set in a compiler command-line flag, and that we do so in the CMakeLists.txt file.